docs/permissions/data-sandboxes-draft.md

```diff
@@ -0,0 +1,309 @@
+---
+title: Data sandboxes
+---
+
+# Data sandboxes
+
+{% include plans-blockquote.html feature="Data sandboxes" %}
+
+Data sandboxes let you give granular permissions to rows and columns for different groups of people.
+
+Say you have people who you want to be able to log into your Metabase, but who should only be able to view data that pertains to them. For example, you might want to display an Accounts table to some customers, but each customer should only be able to view the rows that match their customer ID. Sandboxes let you give people a way to self-serve their own analytics, without ever seeing any results that don't pertain to them.
+
+## How sandboxes work
+
+Data sandboxes work by displaying an edited version of a table, instead of the original table, to a specific group.
+
+You can think of a data sandbox as a bundle of permissions that includes:
+
+- The edited version of a table that will replace the original table, everywhere that the original table is used in Metabase.
+- The [group](../people-and-groups/managing.md#groups) of people who should see the edited version of the table.
+
+You can define up to one data sandbox for each table/group combo in your Metabase. That means you can display different versions of a table for different groups, such as "Sandboxed Accounts for A" to Customer A, and "Sandboxed Accounts for B" to Customer B.
+
+## Types of data sandboxes
+
+A [row-restricted sandbox](#row-restricted-sandboxes) will display a filtered table in place of the original table (to a specific group).
+
+An [advanced sandbox](#advanced-sandboxes) will display a custom query result in place of the original table (to a specific group).
+
+### Row-restricted sandboxes
+
+A row-restricted sandbox displays the result of a filtered table based on a [user attribute](../people-and-groups/managing.md#adding-user-attributes).
+
+A group assigned to a row-restricted sandbox will see a table with some `Filter = "User Attribute"` applied.
+
+For example, you can create a row-restricted sandbox to filter an Accounts table for a sandboxed group so that:
+- A sandboxed person with the user attribute "Basic" will see the rows where `Plan = "Basic"`.
+- A sandboxed person with the user attribute "Premium" will only see the rows where `Plan = "Premium"`.
+
+For more info, see [Choosing user attributes for sandboxing](#choosing-user-attributes-for-sandboxing).
+
+### Advanced sandboxes
```

Comment:
What about "custom sandbox"?

I'm still trying to avoid boxing in (ha) the second sandbox type. 

It isn't quite a "column sandbox" (because you can restrict rows), but it isn't strictly "row and column-restricting sandbox" either (because you can edit a column instead of excluding it). 

You can also create an advanced sandbox JUST for the sake of passing user attributes to parameters in Metabase. (I'm thinking of the [Markdown URL](https://www.metabase.com/learn/dashboards/markdown#create-a-custom-url-with-a-filter-value) use case). 

Of course, it is very useful for people to access user attributes from parameters without necessarily enforcing permissions. But you cannot currently pass a user attribute directly to a parameter in Metabase -- you have to set up a sandbox in the middle. (A success eng candidate actually pointed that out to us recently). In that case, it seems easier to tell people to create a "custom sandbox" that doesn't apply any permissions -- just `SELECT *` from table.

All of this to say that I don't know if parameter functionality will evolve faster than sandboxes (unless we have a read on that from product?). So I think we may want to keep the name future-proof for just a bit longer.

Reply:
You're right. The "handle" is important. And I like "custom sandbox", and think it's sufficiently different to merit the change (especially given we only use "Advanced" in the docs/learn). Advanced implies more capable, but can also suggest a higher degree of difficulty; "custom" implies flexibility, tailoring, etc, without implication of requiring a lot of knowledge or expertise.

So how about something like:

```
### Basic data sandboxes (or row-restricted sandboxes, or row-based data sandboxes).

These sandboxes filter by a column in the table based on user attributes.

Use a [NAME] sandbox when you need to:

- Restrict rows based on user attributes
  
### Custom data sandboxes

These sandboxes uses a saved question to create a custom view for tables.

Use a custom data sandbox when you need to:

- Restrict rows based on user attribute
- Restrict columns based on user attributes
- Create custom column views (like only show the username of an email)
- Pass user attributes as parameters... (link to relevant docs)
```

Or whatever. Given our conversation, the constraints for this quick overview seem like they should be:

- List use cases so people know which box to choose.
- Give people a handle to talk about each method.
- Reflect language used in Metabase itself.
- Summarize how they're setup/how they work.

So, rewrite with those constraints in mind.

> You cannot currently pass a user attribute directly to a parameter in Metabase -- you have to set up a sandbox in the middle. (A success eng candidate actually pointed that out to us recently). In that case, it seems easier to tell people to create a "custom sandbox" that doesn't apply any permissions -- just SELECT * from table.

Is the basic, filter-by-column sandbox insufficient for this? Or is the idea that you want to be able to pass parameters _without_ restricting views?

Reply:
> Is the basic, filter-by-column sandbox insufficient for this? Or is the idea that you want to be able to pass parameters without restricting views?

Yeah, the second thing! 

You can imagine -- it is super useful to be able to access a user attribute from a parameter. For now, it would be great to show people that they can link those things together (via a sandbox), even if they have no need for the permissions part.